File: C++/55JumpGame.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
       int destinationIndex = nums.size() - 1;
       // start from the back and see if you can make it to dest
       for(int i = nums.size() - 2; i >= 0; --i){
            if(nums[i] >= destinationIndex - i){
                destinationIndex = i;
            }
       }
       return destinationIndex == 0;
    }
};

File: Swift/55JumpGame.swift
class Solution {
    func canJump(_ nums: [Int]) -> Bool {
        var farthestReach = 0
        for (i, num) in nums.enumerated() {
            if i > farthestReach {
                return false
            }
            farthestReach = max(num + i, farthestReach)
        }

        return true
    }
}
